src/qt/navcoin.cpp

```diff
@@ -333,13 +339,39 @@ void NavCoinApplication::createOptionsModel(bool resetSettings)
     optionsModel = new OptionsModel(NULL, resetSettings);
 }
 
-void NavCoinApplication::createWindow(const NetworkStyle *networkStyle)
+// this will be used to get mnemonic words
+bool NavCoinApplication::setupMnemonicWords(std::string& wordlist) {
+    namespace fs = boost::filesystem;
+    if (GetBoolArg("-disablewallet", false)) {
+        LogPrintf("Wallet disabled!\n");
+    }
+
+    if (GetBoolArg("-skipmnemonicbackup",false)) {
+        return true;
+    }
+
+    std::string walletFile = GetArg("-wallet", "wallet.dat");
+    if (fs::exists(walletFile)) return true;
+	
+	if (CheckIfWalletDatExists()) return true;
+
+    StartOptionsMain dlg(nullptr);
+    dlg.exec();
+    wordlist = dlg.getWords();
+    return false;
+}
+
+bool NavCoinApplication::createWindow(const NetworkStyle *networkStyle)
 {
+    if (!setupMnemonicWords(wordlist)) {
+        if (wordlist.empty()) return false;
+    }
+
     window = new NavCoinGUI(platformStyle, networkStyle, 0);
 
     pollShutdownTimer = new QTimer(window);
     connect(pollShutdownTimer, SIGNAL(timeout()), window, SLOT(detectShutdown()));
-    pollShutdownTimer->start(200);
```

Comment:
what are the consequences of removing this?

Reply:
My bad I will add that back, when I was porting it, looked like I should remove it. It was moved in Upstream bitcoin so I guess I must have been a little confused or something when I did that, am gonna add it back now.

Reply:
Ok that changes is now reverted